src/crypto_utils.ts

```diff
@@ -157,14 +160,73 @@ export async function buildStringInputText(
 
     let encodedStr = encoder.encode(plaintext)
 
-    let encryptedStr = new Array<{ ciphertext: bigint, signature: Uint8Array }>(plaintext.length)
+    const inputText = {
+        ciphertext: { value: new Array<bigint> },
+        signature: new Array<Uint8Array>
+    }
+
+    for (let i = 0; i < encodedStr.length / 8; i++) {
+        const startIdx = i * 8
+        const endIdx = Math.min(startIdx + 8, encodedStr.length)
+
+        const byteArr = new Uint8Array([...encodedStr.slice(startIdx, endIdx), ...new Uint8Array(8 - (endIdx - startIdx))])
+
+        const it = buildInputText(
+            decodeUint(byteArr),
+            sender,
+            contractAddress,
+            functionSelector
+        )
+
+        inputText.ciphertext.value.push(it.ciphertext)
+        inputText.signature.push(it.signature)
+    }
+
+    return inputText
+}
 
-    for (let i = 0; i < plaintext.length; i++) {
-        const {ctInt, signature} = buildInputText(BigInt(encodedStr[i]), sender, contractAddress, functionSelector)
-        encryptedStr[i] = {ciphertext: ctInt, signature}
+export function buildAddressInputText(
+    plaintext: string,
+    sender: { wallet: BaseWallet; userKey: string },
+    contractAddress: string,
+    functionSelector: string
+) {
+    // consider improving address validation
+    if (plaintext.substring(0, 2) !== '0x' || plaintext.length !== 42) {
+        throw new Error("Invalid address")
     }
 
-    return encryptedStr
+    const it1 = buildInputText(
+        BigInt("0x" + plaintext.substring(2, 18)), // bytes 1 - 8
+        sender,
+        contractAddress,
+        functionSelector
+    )
+    const it2 = buildInputText(
+        BigInt("0x" + plaintext.substring(18, 34)), // bytes 9 - 16
+        sender,
+        contractAddress,
+        functionSelector
+    )
+    const it3 = buildInputText(
+        BigInt("0x" + plaintext.substring(34, 42)), // bytes 17 - 20
+        sender,
+        contractAddress,
+        functionSelector
+    )
+
+   const inputText = {
```

Comment:
i think we should enhance it to a datatype of some ciphered address to manage the 3 parts away from the user

Reply:
Do you mean something like [this](https://github.com/coti-io/coti-sdk-typescript/blob/smiller-coti/coti-ethers/src/types.ts)?